File: week01/BestTimetobuyandsellStock2.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let left = 0;
  let right = 1;
  let result = 0;

  while (left < prices.length && right < prices.length) {
    if (prices[left] > prices[right]) {
      left++;
    } else {
      result = Math.max(result, prices[right] - prices[left]);
      right++;
    }
  }
  return result;
};

File: week01/best_time_to_buy_and_sell_stock_2.cpp
// TC: O(N)
// MC: O(1)
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int mx = 0;
        for(int i = 0; i + 1 < p.size(); i++)
            if(p[i + 1] > p[i])
                mx += p[i + 1] - p[i];
        return mx;
    }
};